sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java

```diff
@@ -748,87 +699,85 @@ private String createZipFile(List<String> expected, @Nullable String filename, S
     return tmpFileName;
   }
 
-  /**
-   * Read a zip compressed file. The user provides the ZIP compression type.
-   * We expect a PCollection with the lines.
-   */
   @Test
   @Category(NeedsRunner.class)
-  public void testZipCompressedRead() throws Exception {
-    String[] lines = {"Irritable eagle", "Optimistic jay", "Fanciful hawk"};
-    List<String> expected = new ArrayList<>();
-
-    String filename = createZipFile(expected, null, lines);
+  public void testTxtReadRead() throws Exception {
+    // Files with non-compressed extensions should work in AUTO and UNCOMPRESSED modes.
+    for (CompressionType type : new CompressionType[] { AUTO, UNCOMPRESSED }) {
+      assertReadingCompressedFileMatchesExpected(emptyTxt, type, EMPTY);
+      assertReadingCompressedFileMatchesExpected(tinyTxt, type, TINY);
+      assertReadingCompressedFileMatchesExpected(largeTxt, type, LARGE);
+    }
+  }
 
-    Pipeline p = TestPipeline.create();
+  @Test
+  @Category(NeedsRunner.class)
+  public void testGzipCompressedRead() throws Exception {
```

Comment:
Is it possible to further reduce code duplication between testing different compression algorithms?